apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt

```diff
@@ -70,47 +64,122 @@ open class ChatController(val ctrl: ChatCtrl) {
   suspend fun apiGetActiveUser(): User? {
     val r = sendCmd(CC.ShowActiveUser())
     if (r is CR.ActiveUser) return r.user
-    Log.d("SIMPLEX", "apiGetActiveUser: ${r.toString()}")
+    Log.d("SIMPLEX", "apiGetActiveUser: ${r.responseType} ${r.details}")
     return null
   }
 
   suspend fun apiCreateActiveUser(p: Profile): User {
     val r = sendCmd(CC.CreateActiveUser(p))
     if (r is CR.ActiveUser) return r.user
-    Log.d("SIMPLEX", "apiCreateActiveUser: ${r.toString()}")
-    throw Error("user not created ${r.toString()}")
+    Log.d("SIMPLEX", "apiCreateActiveUser: ${r.responseType} ${r.details}")
+    throw Error("user not created ${r.responseType} ${r.details}")
   }
 
   suspend fun apiStartChat() {
     val r = sendCmd(CC.StartChat())
     if (r is CR.ChatStarted ) return
-    throw Error("failed starting chat: ${r.toString()}")
+    throw Error("failed starting chat: ${r.responseType} ${r.details}")
   }
 
   suspend fun apiGetChats(): List<Chat> {
     val r = sendCmd(CC.ApiGetChats())
     if (r is CR.ApiChats ) return r.chats
-    throw Error("failed getting the list of chats: ${r.toString()}")
+    throw Error("failed getting the list of chats: ${r.responseType} ${r.details}")
   }
 
   suspend fun apiGetChat(type: ChatType, id: Long): Chat? {
     val r = sendCmd(CC.ApiGetChat(type, id))
     if (r is CR.ApiChat ) return r.chat
-    Log.d("SIMPLEX", "apiGetChat bad response: ${r.toString()}")
+    Log.d("SIMPLEX", "apiGetChat bad response: ${r.responseType} ${r.details}")
     return null
   }
 
   suspend fun apiSendMessage(type: ChatType, id: Long, mc: MsgContent): AChatItem? {
     val r = sendCmd(CC.ApiSendMessage(type, id, mc))
     if (r is CR.NewChatItem ) return r.chatItem
-    Log.d("SIMPLEX", "apiSendMessage bad response: ${r.toString()}")
+    Log.d("SIMPLEX", "apiSendMessage bad response: ${r.responseType} ${r.details}")
+    return null
+  }
+
+  suspend fun apiAddContact(): String? {
```

Comment:
Why does this not take any arguments? It feels like it should take some identifier of the contact to be added

Reply:
At the time it's created you don't know it. We could consider giving names to this invitations, but now the name arrives in the profile when the invitation is accepted by the other party

Reply:
These invitations links will expire after 24-48 hours.